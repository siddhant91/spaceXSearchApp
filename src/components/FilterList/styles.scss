@import '../../styles/main.scss';

.spacex-filter-list {
	&--title {
		@include padding(0, 0, $spacex-spacing-1);
		border-bottom: 1px solid $spacex-color-black;
		@include margin($spacex-spacing-2);
		text-align: center;
		@media only screen and (min-width: $spacex-breakpoint-x-large) {
			@include margin($spacex-spacing-2, $spacex-spacing-5, 0, $spacex-spacing-5);
		}
	}
	&--items {
		@include margin($spacex-spacing-3, 0, $spacex-spacing-3, 0);
	}
}
