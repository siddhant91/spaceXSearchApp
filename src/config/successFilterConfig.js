export default [
	{
		text: 'True',
		value: true,
		isSelected: false,
	},
	{
		text: 'False',
		value: false,
		isSelected: false,
	},
];
