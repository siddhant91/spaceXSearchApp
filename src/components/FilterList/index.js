import { arrayOf, string, shape, oneOfType, bool, number, func } from 'prop-types';

// Components
import Button from '../Button';

// Styles
import './styles.scss';

const FilterList = ({
	filterTitle,
	filterList,
	filterKey,
	updateFilterList,
	selectedFiltersList,
}) => {
	const handleFilterClick = (e) => {
		const { value } = e.target.dataset;
		updateFilterList({
			filterKey,
			value,
		});
	};
	const getFilters = () => {
		if (filterList.length) {
			const filters = filterList.map((filter) => {
				const { text, value } = filter;
				const isSelected = selectedFiltersList.indexOf(value.toString()) > -1;
				return (
					<Button
						handleClick={handleFilterClick}
						data-value={value}
						isSelected={isSelected}
						key={`${filterKey}-${value}`}
						aria-label={text}
						data-testid="filter-button"
					>
						{text}
					</Button>
				);
			});
			return filters;
		}
		return null;
	};

	return (
		<div className="spacex-filter-list">
			<div className="spacex-filter-list--title">
				<h3>{filterTitle}</h3>
			</div>
			<div
				className="d-flex flex-wrap justify-content-between spacex-filter-list--items"
				data-testid="filter-list-container"
			>
				{getFilters()}
			</div>
		</div>
	);
};

FilterList.propTypes = {
	filterTitle: string.isRequired,
	filterList: arrayOf(
		shape({
			text: string,
			value: oneOfType([bool, number]),
			isSelected: bool,
		}),
	).isRequired,
	updateFilterList: func.isRequired,
	filterKey: string.isRequired,
	selectedFiltersList: arrayOf(string),
};

FilterList.defaultProps = {
	selectedFiltersList: [],
};

export default FilterList;
