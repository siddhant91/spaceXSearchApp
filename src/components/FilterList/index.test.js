import { render, cleanup, screen } from '@testing-library/react';

import FilterList from './index';
import successFilterConfig from '../../config/successFilterConfig';

afterEach(cleanup);

const fn = jest.fn();
const renderComponent = ({ filterList = [] } = {}) => {
	return (
		<FilterList
			filterTitle="Successful Landing"
			filterKey="landing_success"
			filterList={filterList}
			updateFilterList={fn}
			selectedFiltersList={[]}
		/>
	);
};

it('Matches snapshot', () => {
	const { asFragment } = render(renderComponent({ filterList: successFilterConfig }));
	expect(asFragment()).toMatchSnapshot();
});

it('renders the Filter title ', async () => {
	const { findByText } = render(renderComponent());
	expect(await findByText('Successful Landing')).toBeInTheDocument();
});

it('renders filter list', async () => {
	render(renderComponent({ filterList: successFilterConfig }));
	const filtersbtns = await screen.findAllByTestId('filter-button');
	expect(filtersbtns).toHaveLength(2);
});

it('renders no filter', async () => {
	render(renderComponent());
	const filtersbtnsContainer = await screen.findByTestId('filter-list-container');
	expect(filtersbtnsContainer).toBeEmptyDOMElement();
});
