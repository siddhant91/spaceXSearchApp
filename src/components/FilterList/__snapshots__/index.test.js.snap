// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spacex-filter-list"
  >
    <div
      class="spacex-filter-list--title"
    >
      <h3>
        Successful Landing
      </h3>
    </div>
    <div
      class="d-flex flex-wrap justify-content-between spacex-filter-list--items"
      data-testid="filter-list-container"
    >
      <button
        aria-label="True"
        class="spacex-button-component"
        data-testid="filter-button"
        data-value="true"
        type="button"
      >
        True
      </button>
      <button
        aria-label="False"
        class="spacex-button-component"
        data-testid="filter-button"
        data-value="false"
        type="button"
      >
        False
      </button>
    </div>
  </div>
</DocumentFragment>
`;
